tmt-app/src/Components/QuizStepper.jsx

```diff
@@ -0,0 +1,124 @@
+import {
+  Box,
+  Container,
+  Step,
+  StepConnector,
+  stepConnectorClasses,
+  StepLabel,
+  Stepper,
+  styled,
+  Typography,
+  useMediaQuery,
+} from '@mui/material';
+import * as React from 'react';
+import { useLocation } from 'react-router-dom';
+
+import HomeIcon from '../Images/icons/boba_progress/Home.svg';
+import OrderUpIcon from '../Images/icons/boba_progress/OrderUp.svg';
+import SoftSkillsIcon from '../Images/icons/boba_progress/SoftSkills.svg';
+import SubjectsIcon from '../Images/icons/boba_progress/Subjects.svg';
+
+const ProgressConnector = styled(StepConnector)(({ theme }) => ({
+  [`&.${stepConnectorClasses.active}`]: {
+    [`& .${stepConnectorClasses.line}`]: {
+      backgroundColor: theme.palette.Text.main,
+    },
+  },
+  [`&.${stepConnectorClasses.completed}`]: {
+    [`& .${stepConnectorClasses.line}`]: {
+      backgroundColor: theme.palette.Text.main,
+    },
+  },
+  [`& .${stepConnectorClasses.line}`]: {
+    backgroundColor: theme.palette.BobaHighlight.main,
+    border: 0,
+    borderRadius: 0,
+    height: 12,
+  },
+}));
+
+function ProgressIcon(props) {
+  const icons = [
+    {
+      icon: <img width="64px" src={HomeIcon} alt="Home" />,
+      smallText: 'Leave the Shop',
+      largeText: 'Home',
+    },
+    {
+      icon: <img width="64px" src={SubjectsIcon} alt="Subjects" />,
+      smallText: 'Pick a Base',
+      largeText: 'Subjects',
+    },
+    {
+      icon: <img width="64px" src={SoftSkillsIcon} alt="Soft Skills" />,
+      smallText: 'Pick Tapioca',
+      largeText: 'Soft Skills',
+    },
+    {
+      icon: <img width="64px" src={OrderUpIcon} alt="Order Up" />,
+      smallText: 'Sealed the drink',
+      largeText: 'Order Up!',
+    },
+  ];
+
+  const { icon, smallText, largeText } = icons[props.icon - 1];
+
+  const isSmallScreen = useMediaQuery((theme) => theme.breakpoints.down('sm'));
+
+  return (
+    <Container
+      sx={{
+        alignItems: 'center',
+        display: 'flex',
+        flexDirection: 'column',
+        height: 50,
+        justifyContent: 'center',
+        overflow: 'visible',
+        zIndex: 1,
+      }}
+    >
+      {isSmallScreen ? (
+        <div>{icon}</div>
+      ) : (
+        <>
+          <Typography variant="caption" sx={{ textAlign: 'center' }}>
+            {smallText}
+          </Typography>
+          <div>{icon}</div>
+          <Typography
+            variant="subtitle1"
+            sx={{ fontWeight: 'bold', textAlign: 'center' }}
+          >
+            {largeText}
+          </Typography>
+        </>
+      )}
+    </Container>
+  );
+}
+
+export default function QuizStepper() {
+  const location = useLocation();
+  const routes = {
+    '/': 0,
+    '/quiz/classes': 1,
+    '/quiz/skills': 2,
+    '/receipt': 3,
+  };
```

Comment:
if we consider having all of the steps be parameterized, ie using `/quiz` as a base route, we could look at the route structure as `/quiz/:step` and use `:step` as the key for these objects rather than resort to a static number and array index. It would probably be much more intuitive in the long run

Reply:
@jwu910 Sorry for the long inactivity been out of the country for a bit! But to clarify when you mean parameterizing the routes what exactly does that mean? I am a bit confused if you're saying I should still use a map and the route names as keys or if you want me to do something else.

Reply:
Oh, I think you should use the url parameters via react router